### PR TITLE
21 create the classes management tab content in the admin dashboard

### LINKED ISSUE #21: create the Classes management tab content in the admin dashboard

### PR DESCRIPTION
**Pull request by iyassalah**
this is ready for QA, @yamenw

### CODE REVIEW

File: src/components/class-management/class-management.tsx [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+
+function ClassManagement() {
+  const [classes, setClasses] = useState<IClass[]>([]);
+  const [loading, setLoading] = useState(false);
+  const { token } = useContext(AuthContext);
+
+  useEffect(() => {
+    setLoading(true);
+    axios
+      .get("/classes")
```
Comment (yamenw) [inline src/components/class-management/class-management.tsx]:
```suggestion
    axios
      .get<Responses.IGetCLasses("/classes")
```
Add a type for this response in `responses.ts`

File: src/components/class-management/class-management.tsx [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+    },
+    {
+      title: "Displayed Name",
+      dataIndex: "displayedName",
+      key: "displayedName",
+      render: (text: string, record: IClass) => (
+        <Input
+          defaultValue={text}
+          onBlur={(e) =>
+            handleUpdateDisplayedName(record.internalName, e.target.value, token)
```
Comment (yamenw) [inline src/components/class-management/class-management.tsx:67]: You have a type error here, if the token is `null`, then you should redirect the user out of here.

File: src/components/class-management/class-management.tsx [hunk 3]
```diff
@@ -0,0 +1,89 @@
+import { useContext, useEffect, useState } from "react";
+import { Button, Form, Input, message, Table } from "antd";
```
Comment (yamenw) [inline src/components/class-management/class-management.tsx]: Unused imports

**Review by yamenw — CHANGES_REQUESTED** [review]
Thanks for submitting this, please address my comments so that we can merge this.

File: src/components/class-management/class-management.tsx (hunk 3 above)
Comment (iyassalah) [reply src/components/class-management/class-management.tsx]: removed

File: src/components/class-management/class-management.tsx (hunk 2 above)
Comment (iyassalah) [reply src/components/class-management/class-management.tsx:67]: man do this please, I dont know why it dosent work with me

**Review by yamenw — APPROVED** [review]
Just merge it, well fix the login in another ticket

**Pull request merged**